src/main/java/jnr/unixsocket/UnixSocket.java

```diff
@@ -14,45 +14,147 @@
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
+ * 
+ * ported from  https://github.com/softprops/unisockets/blob/master/unisockets-core/src/main/scala/Socket.scala 
  */
-
 package jnr.unixsocket;
 
-import jnr.constants.platform.SocketLevel;
-import jnr.constants.platform.SocketOption;
-import jnr.enxio.channels.NativeSocketChannel;
-import jnr.ffi.byref.IntByReference;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.InetAddress;
+import java.net.SocketAddress;
+import java.nio.channels.Channels;
+import java.nio.channels.SocketChannel;
 
-import java.nio.ByteBuffer;
-import java.nio.ByteOrder;
-import java.nio.channels.Channel;
+class UnixSocket extends java.net.Socket {
 
-public class UnixSocket {
-    private final NativeSocketChannel channel;
+	private UnixSocketChannel chan;
```

Comment:
You need to remove all tabs from this PR. We follow standard Java indentation of four spaces.